nyc/bills.py

```diff
@@ -1,232 +1,359 @@
 
-    SESSION_STARTS = (2014, 2010, 2006, 2002, 1996)
+        # Add required URL parameter
+        self.params = {'Token': TOKEN}
 
-    def sessions(self, action_date) :
-        for session in self.SESSION_STARTS :
-            if action_date >= datetime.datetime(session, 1, 1, 
-                                               tzinfo=pytz.timezone(self.TIMEZONE)) :
+
+    def sessions(self, action_date):
+        for session in self.SESSION_STARTS:
+            if action_date >= datetime.datetime(session, 1, 1,
+                                               tzinfo=pytz.timezone(self.TIMEZONE)):
                 return str(session)
 
 
+    def actions(self, matter_id):
+        for action in self.history(matter_id):
+            bill_action = {}
+
+            if not action['MatterHistoryActionName']:
+                continue
+
+            bill_action['action_description'] = action['MatterHistoryActionName']
+            bill_action['action_date'] = action['MatterHistoryActionDate']
+
+            responsible_org = action['MatterHistoryActionBodyName']
+
+            if responsible_org == 'City Council':
+                responsible_org = 'New York City Council'
+
+            elif responsible_org == 'Administration':
+                responsible_org = 'Mayor'
+
+            elif responsible_org == 'Town Hall Meeting':
+                continue
+
+            bill_action['responsible_org'] = responsible_org
+
+            if all(bill_action.values()):
+                action_date = self.toTime(bill_action['action_date']).date()
+                bill_action['action_date'] = action_date
+
+                # Strip trailing newline and carriage return chars
+                classification = ACTION_CLASSIFICATION[bill_action['action_description'].strip()]
+                bill_action['classification'] = classification
+
+                if all(action.get(k, None) for k in ['MatterHistoryEventId',
+                                                     'MatterHistoryRollCallFlag',
+                                                     'MatterHistoryPassedFlag']):
+
+                    bool_result = action['MatterHistoryPassedFlag']
+                    result = 'pass' if bool_result else 'fail'
+
+                    votes = (result, self.votes(action['MatterHistoryId']))
+                else:
+                    votes = (None, [])
+
+                yield bill_action, votes
+
+
+    def _version_rank(self, version):
+        '''
+        - If there is only one version, *, it is the max version.
+        - If there is an asterisk version and one other version, the other
+          version is the max version.
+        - If there is an asterisk version and two other non-numeric versions,
+          the max version is the first other version in alphabetical order.
+        - Otherwise, function as normal.
+        '''
+        version_map = {'*': -1,
+                       'B': 1,
+                       'A': 2}
```

Comment:
B is before A?

Reply:
Yes, as far as I can tell, if the versions are lettered rather than numbered, the first letter alphabetically is the "max" version, and so should have a higher integer value. [I have an issue open in **nyc-council-councilmatic** for clarification on versioning](https://github.com/datamade/nyc-council-councilmatic/issues/5).